web/packages/core/src/load-options.ts

```diff
@@ -338,6 +338,14 @@ export interface BaseLoadOptions {
      */
     contextMenu?: boolean;
 
+    /**
+     * Whether or not to show a context menu when long-pressing
+     * a Ruffle instance on touch devices.
+     *
+     * @default true
+     */
+    touchContextMenu?: boolean;
```

Comment:
Quite debatable, but my own preference is to avoid introduction of new configs where possible. Rather, I prefer re-usage of existing ones. That's mainly in order to keep Ruffle's API simple and clear to external embedders.

In this case, I'd suggest expanding the existing `contextMenu` config to be an enum. Previous boolean values can be kept for backward compatibility. Something like:
```ts
export interface BaseLoadOptions {
    ...
    contextMenu?: ContextMenu;
    ...
}

export const enum ContextMenu {
    Always = "always",
    AlwaysBackwardsCompatibility = true,

    NoTouch = "noTouch",

    Never = "never",
    NeverBackwardsCompatibility = false,
}
```
The terminology might not fit exactly, I'm open to ideas (perhaps on/off, enable/disable). Anyway I hope this demonstrates the point in general.

What do you think?

Reply:
I'd be a bit more keen if we could get rid of the old options, but obviously that's not possible due to backwards compatibility. Making `true` and `"always"` function the same way, as well as `false` and `"never"`, will complicate the code a bit, but I suppose it's fine, so I can do that.

In terms of terminology, what about `"on"`, `"off"`, and `"rightClickOnly"`? I think I prefer a more direct "this is when the context menu will appear" naming scheme as opposed to the option defining "this is when the context menu won't appear".

While adding these options, should I add `"touchOnly"`, or do we think that would never be useful?

Reply:
`const enum member initializers must be constant expressions`

Reply:
https://github.com/microsoft/TypeScript/issues/32192#issuecomment-507205898

Reply:
I like your suggested terminology, guess we should proceed with it.

Regarding backwards compatibility, as already done on previous API changes, we can `console.warn()` when the boolean values are used, in hope that embedders will eventually migrate to the new values, and after a grace period we will be able to remove the old ones.

Reply:
Enums cannot contain booleans, so I don't know if this can be done.

Reply:
Hmm, in that case we can define `contextMenu?: ContextMenu | boolean;`, and omit the `BackwardsCompatibility` members from the `ContextMenu` enum. Alternatively, we can take inspiration from https://github.com/microsoft/TypeScript/issues/32192#issuecomment-507484261, but that feels to hacky to me.

Reply:
Done. Something else occurred to me while working on this. The `checkIfTouch` function should have an `else`, `this.isTouch = false`, right? That way a device with touch support and mouse support will know that a mouse was used to open the context menu even if the first time the context menu was opened was done by touch.

Reply:
If so, should I add that to this PR?